Purchases/Misc/Obsoletions.swift

```diff
@@ -395,6 +395,43 @@ public extension Purchases {
         fatalError()
     }
 
+    /**
+     * Use this method to find eligibility for this user for
+     * [iOS Promotional Offers](https://docs.revenuecat.com/docs/ios-subscription-offers#promotional-offers).
+     * - Note: If you're looking to use free trials or Introductory Offers instead,
+     * use ``Purchases/checkTrialOrIntroDiscountEligibility(productIdentifiers:completion:)``.
+     *
+     * - Parameter discount: The ``StoreProductDiscount`` to apply to the product.
+     * - Parameter product: The ``StoreProduct`` the user intends to purchase.
+     */
+    @available(iOS, introduced: 13.0, unavailable, renamed: "promotionalOffer(forProductDiscount:product:)")
+    @available(tvOS, introduced: 13.0, unavailable, renamed: "promotionalOffer(forProductDiscount:product:)")
+    @available(watchOS, introduced: 6.2, unavailable, renamed: "promotionalOffer(forProductDiscount:product:)")
+    @available(macOS, introduced: 10.15, unavailable, renamed: "promotionalOffer(forProductDiscount:product:)")
+    @available(macCatalyst, introduced: 13.0, unavailable, renamed: "promotionalOffer(forProductDiscount:product:)")
+    func getPromotionalOffer(forProductDiscount discount: StoreProductDiscount,
+                             product: StoreProduct) async throws -> PromotionalOffer {
+        fatalError()
+    }
```

Comment:
I think these should be deprecations - if we straight up do Obsoletions, that would mean doing another major version. 
We could have them be deprecations instead (see Deprecations.swift), and have them redirect to the old methods. 
That way they don't require any extra effort for maintaining since they're essentially an alias for existing logic, but they also don't break compilation for existing apps. 
With a deprecation Xcode will still suggest the auto fix-it, but it will continue to work for existing apps using the methods pre-rename